internal/mode/static/policies/policy.go

```diff
@@ -21,7 +23,13 @@ type Policy interface {
 //
 //counterfeiter:generate . ConfigGenerator
 type ConfigGenerator interface {
-	Generate(policy Policy) []byte
+	Generate(policy Policy, globalSettings *ngfAPI.NginxProxy) []byte
+}
+
+// GlobalPolicySettings are settings from the current state of the graph that may be
+// needed for policy validation if certain policies rely on those global settings.
+type GlobalPolicySettings struct {
```

Comment:
perhaps we can rename it to ValidationContext or something like that?
the current name clashes with Generate methods, which uses NginxProxy as globalSettings
```
	Generate(policy Policy, globalSettings *ngfAPI.NginxProxy) []byte
```

Reply:
Why does Generate use NginxProxy instead of GlobalSettings? Ideally, it would also use GlobalSettings.

Reply:
I was thinking that GlobalSettings (or a different name like PolicyContext) would be a subset of the NginxProxy fields that affect policies. By creating this new type we are explicitly listing which fields policies are dependent on. 

```
type GlobalSettings struct {
    TracingEnabled bool
    SpanAttrs map[string]string
}
```

or:

```
type GlobalSettings struct {
    Tracing GlobalSettingsTracing
}

type GlobalSettingsTracing struct {
    Enabled bool
    SpanAttrs map[string]string
} 
```

Reply:
This is because Validate and Generate are called in two different places, and the `GlobalSettings` struct doesn't persist. Validate is called when the Graph is built, whereas Generate is called later on, and the GlobalSettings struct isn't saved on the Graph. I could rebuild it again, but I figured we already have the configuration we need for Generation which lives in the NginxProxy resource in the Graph.

Reply:
I see. Why not store the GlobalSettings on the graph?

Reply:
I just figured that as of right now the data we need for Generation already exists in the NginxProxy resource, so why add another struct with the same info.

Reply:
I think code duplication is acceptable in cases with a clear benefit. 

The NginxProxy resource is a superset of the fields on which the policies depend. We are passing extra information to the Generator that's different from the information we are passing to the validator. This creates a dependency between the NginxProxy and the policy generator. Say we change the telemetry structure in the NginxProxy or decide that the telemetry fields belong in a separate policy. We will have to update the observability policy's generator and the code that creates the GlobalSettings.

I think we want the dependency to be on the settings that can affect policies, not the object that contains these settings.

@pleshakov thoughts?

Reply:
> The NginxProxy resource is a superset of the fields on which the policies depend. We are passing extra information to the Generator that's different from the information we are passing to the validator. This creates a dependency between the NginxProxy and the policy generator.

that makes sense to me